workers/data_refinery_workers/downloaders/test_array_express.py

```diff
@@ -0,0 +1,117 @@
+import copy
+from unittest.mock import patch, MagicMock
+from django.test import TestCase
+from data_refinery_models.models import (
+    SurveyJob,
+    Batch,
+    BatchStatuses,
+    DownloaderJob,
+    DownloaderJobsToBatches,
+    ProcessorJob,
+    ProcessorJobsToBatches
+)
+from data_refinery_workers.downloaders import array_express
+
+
+class DownloadArrayExpressTestCase(TestCase):
+    def test_good_batch_grouping(self):
+        """Returns true if all batches have the same download_url."""
+        batches = [Batch(download_url="https://example.com"),
+                   Batch(download_url="https://example.com"),
+                   Batch(download_url="https://example.com")]
+        job_id = 1
+        self.assertIsNone(array_express._verify_batch_grouping(batches, job_id))
+
+    def test_bad_batch_grouping(self):
+        """Raises exception if all batches don't have the same download_url."""
+        batches = [Batch(download_url="https://example.com"),
+                   Batch(download_url="https://example.com"),
+                   Batch(download_url="https://wompwomp.com")]
```

Comment:
Only for 1800 bucks lol. `sadtrombone.com` is great though, it's everything I'd ever want that site to be.